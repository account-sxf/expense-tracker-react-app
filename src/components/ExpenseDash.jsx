import React from "react"
import ExpenseList from "./ExpenseList"
import ExpenseListFilter from "./ExpenseListFilter"

const ExpenseDashboardPage = (props) => (
  <div>
    <div>
      <h2>
        This is from my dashboard
      </h2>
      <ExpenseListFilter />
      <ExpenseList />
    </div>
  </div>
)

export default ExpenseDashboardPage
