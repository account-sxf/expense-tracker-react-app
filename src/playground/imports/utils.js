// export { remove, flatten as default }

/*
  remove([1, 2, 3, 4, 5, 6], [1, 3, 6]); // [2, 4, 5]
*/
export function remove(arr1, arr2) {
  if (!Array.isArray(arr1) || !Array.isArray(arr2)) {
    throw new Error('expected both arguments to be arrays');
  }
  var result = [];
  var len = arr1.length;
  for (var i = 0; i < len; i++) {
    var elem = arr1[i];
    if (arr2.indexOf(elem) == -1) {
      result.push(elem);
    }
  }
  return result;
}

/*
  flatten([[1, [2, 3]], [[4, 5], 6, 7, [8, 9]]]);
  // [1, 2, 3, 4, 5, 6, 7, 8, 9]
*/
export function flattenHelper(arr, depth) {
  var stack = arr.slice();
  var result = [];

  while (stack.length) {
    var item = stack.pop();

    if (Array.isArray(item) && depth > 0) {
      stack.push.apply(stack, item);
      depth--;
    } else {
      result.push(item);
    }
  }

  return result.reverse();
}

function flatten(arr, depth) {
  if (!Array.isArray(arr)) {
    throw new Error('expected an array');
  }

  if (depth !== undefined && typeof depth !== 'number') {
    throw new Error('depth expects a number');
  }

  return flattenHelper(arr, typeof depth === 'number' ? depth : Infinity);
}
export default flatten

/*
  const cubedRoot = partial(Math.pow, _, 1/3);
  cubedRoot(64); // 4
  const getRoot = partial(Math.pow, 64);
  getRoot(1/2); // 8
*/
export function partial(fn /*, arg1, arg2 etc */) {
  var partialArgs = [].slice.call(arguments, 1);

  if (!partialArgs.length) {
    return fn;
  }

  return function () {
    var args = [].slice.call(arguments);
    var derivedArgs = [];

    for (var i = 0; i < partialArgs.length; i++) {
      var thisPartialArg = partialArgs[i];

      derivedArgs[i] =
        thisPartialArg === undefined ? args.shift() : thisPartialArg;
    }

    return fn.apply(this, derivedArgs.concat(args));
  };
}