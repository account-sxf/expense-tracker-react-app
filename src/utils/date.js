const pad = num => num.toString().padStart(2, '0')

export const utcDateOnly = (time) => {
  const dateTime = new Date(time)
  const day = pad(dateTime.getDate())
  const month = pad(dateTime.getMonth() + 1)
  const year = pad(dateTime.getFullYear())
  return `${year}/${month}/${day}`
}

export const utcDateTime = (time) => {
  const dateTime = new Date(time)
  const day = pad(dateTime.getDate())
  const month = pad(dateTime.getMonth() + 1)
  const year = pad(dateTime.getFullYear())
  const hour = pad(dateTime.getHours())
  const mins = pad(dateTime.getMinutes())
  return `${year}/${month}/${day}T${hour}:${mins}`
}

export const startOfMonth = () => {
  const dateTime = new Date()
  const day = '01'
  const month = pad(dateTime.getMonth() + 1)
  const year = pad(dateTime.getFullYear())
  const startOfMonth = new Date(`${year}-${month}-${day}T00:00:00.000Z`)
  return startOfMonth.getTime()
}

export const endOfMonth = () => {
  const dateTime = new Date()
  const day = '01'
  const month = pad(dateTime.getMonth() + 2)
  const year = pad(dateTime.getFullYear())
  const endOfMonth = new Date(`${year}-${month}-${day}T00:00:00.000Z`)
  return endOfMonth.getTime()
}