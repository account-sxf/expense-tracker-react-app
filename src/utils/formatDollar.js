const nf = new Intl.NumberFormat('en-NZ', {
  style: "currency",
  currency: "NZD",
  maximumFractionDigits: 2,
  roundingIncrement: 1
})

export default (cents) => {
  const dollars = cents / 100
  return nf.format(dollars)
}