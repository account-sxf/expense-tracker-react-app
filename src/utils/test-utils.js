import React from 'react'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import createStore from '../store'
import { BrowserRouter } from 'react-router-dom'
import expenses from '../tests/fixtures/expenses'

export function renderWithWrappers(
  ui, {
    // get test data
    preloadedState = { expenses: { items: [...expenses] } },
    // create store with the preloaded test data
    store = createStore(preloadedState),
    ...renderOptions
  } = {}
) {

  // Wrapper for react router and redux for integrated tests
  function Wrapper({ children }) {
    return (
      <BrowserRouter>
        <Provider store={store}>
          {children}
        </Provider>
      </BrowserRouter>
    )
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export function renderWithStoreOnly(
  ui, {
    // get test data
    preloadedState = { expenses: { items: [...expenses] } },
    // create store with the preloaded test data
    store = createStore(preloadedState),
    ...renderOptions
  } = {}
) {

  // Wrapper for react router and redux for integrated tests
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    )
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}