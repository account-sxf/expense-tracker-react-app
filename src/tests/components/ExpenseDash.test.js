/**
 * @jest-environment jsdom
 */
import React from "react"
import '@testing-library/jest-dom'
import ExpenseDash from "../../components/ExpenseDash"
import { renderWithWrappers } from '../../utils/test-utils'

test("Test expense list item", () => {
  const { asFragment } = renderWithWrappers(<ExpenseDash />)
  expect(asFragment()).toMatchSnapshot()
})