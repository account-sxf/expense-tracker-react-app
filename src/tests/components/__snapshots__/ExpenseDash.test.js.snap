// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test expense list item 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h2>
        This is from my dashboard
      </h2>
      <div>
        <input
          type="text"
          value=""
        />
        <select
          data-testid="select"
        >
          <option
            data-testid="select-option"
            value="date"
          >
            Date
          </option>
          <option
            data-testid="select-option"
            value="amount"
          >
            Amount
          </option>
          <option
            data-testid="select-option"
            value="description"
          >
            Description
          </option>
        </select>
        <button>
          Descending
        </button>
        <div
          class="react-daterange-picker react-daterange-picker--closed react-daterange-picker--enabled"
          style="position: relative;"
        >
          <div
            class="react-daterange-picker__wrapper"
          >
            <div
              class="react-daterange-picker__inputGroup"
            >
              <input
                aria-label="Date Range"
                max="275760-09-13"
                min="0001-01-01"
                name="daterange_from"
                style="visibility: hidden; position: absolute; z-index: -999;"
                type="date"
                value=""
              />
              <input
                autocomplete="off"
                class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__year"
                data-input="true"
                inputmode="numeric"
                max="275760"
                min="1"
                name="year"
                placeholder="----"
                step="1"
                style="width: 16px;"
                type="number"
                value=""
              />
              <span
                class="react-date-picker__inputGroup__divider"
              >
                /
              </span>
              <input
                aria-label="Month"
                autocomplete="off"
                class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__month"
                data-input="true"
                inputmode="numeric"
                max="12"
                min="1"
                name="month"
                placeholder="--"
                style="width: 8px;"
                type="number"
                value=""
              />
              <span
                class="react-date-picker__inputGroup__divider"
              >
                /
              </span>
              <input
                aria-label="Day"
                autocomplete="off"
                class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__day"
                data-input="true"
                inputmode="numeric"
                max="31"
                min="1"
                name="day"
                placeholder="--"
                style="width: 8px;"
                type="number"
                value=""
              />
            </div>
            <span
              class="react-daterange-picker__range-divider"
            >
              –
            </span>
            <div
              class="react-daterange-picker__inputGroup"
            >
              <input
                aria-label="Date Range"
                max="275760-09-13"
                min="0001-01-01"
                name="daterange_to"
                style="visibility: hidden; position: absolute; z-index: -999;"
                type="date"
                value=""
              />
              <input
                autocomplete="off"
                class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__year"
                data-input="true"
                inputmode="numeric"
                max="275760"
                min="1"
                name="year"
                placeholder="----"
                step="1"
                style="width: 16px;"
                type="number"
                value=""
              />
              <span
                class="react-date-picker__inputGroup__divider"
              >
                /
              </span>
              <input
                aria-label="Month"
                autocomplete="off"
                class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__month"
                data-input="true"
                inputmode="numeric"
                max="12"
                min="1"
                name="month"
                placeholder="--"
                style="width: 8px;"
                type="number"
                value=""
              />
              <span
                class="react-date-picker__inputGroup__divider"
              >
                /
              </span>
              <input
                aria-label="Day"
                autocomplete="off"
                class="react-daterange-picker__inputGroup__input react-daterange-picker__inputGroup__day"
                data-input="true"
                inputmode="numeric"
                max="31"
                min="1"
                name="day"
                placeholder="--"
                style="width: 8px;"
                type="number"
                value=""
              />
            </div>
            <button
              aria-label="Clear value"
              class="react-daterange-picker__clear-button react-daterange-picker__button"
              type="button"
            >
              <svg
                class="react-daterange-picker__clear-button__icon react-daterange-picker__button__icon"
                height="19"
                stroke="black"
                stroke-width="2"
                viewBox="0 0 19 19"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="4"
                  x2="15"
                  y1="4"
                  y2="15"
                />
                <line
                  x1="15"
                  x2="4"
                  y1="4"
                  y2="15"
                />
              </svg>
            </button>
            <button
              aria-label="Toggle calendar"
              class="react-daterange-picker__calendar-button react-daterange-picker__button"
              type="button"
            >
              <svg
                class="react-daterange-picker__calendar-button__icon react-daterange-picker__button__icon"
                height="19"
                stroke="black"
                stroke-width="2"
                viewBox="0 0 19 19"
                width="19"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="15"
                  width="15"
                  x="2"
                  y="2"
                />
                <line
                  x1="6"
                  x2="6"
                  y1="0"
                  y2="4"
                />
                <line
                  x1="13"
                  x2="13"
                  y1="0"
                  y2="4"
                />
              </svg>
            </button>
          </div>
          <div
            class="react-daterange-picker__calendar react-daterange-picker__calendar--closed"
            style="position: absolute; height: 0px; top: 100%; width: 0px; left: 0px;"
          >
            <div
              class="react-calendar"
            >
              <div
                class="react-calendar__navigation"
              >
                <button
                  aria-label=""
                  class="react-calendar__navigation__arrow react-calendar__navigation__prev2-button"
                  type="button"
                >
                  «
                </button>
                <button
                  aria-label=""
                  class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
                  type="button"
                >
                  ‹
                </button>
                <button
                  aria-label=""
                  class="react-calendar__navigation__label"
                  style="flex-grow: 1;"
                  type="button"
                >
                  <span
                    class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
                  >
                    August 2022
                  </span>
                </button>
                <button
                  aria-label=""
                  class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
                  type="button"
                >
                  ›
                </button>
                <button
                  aria-label=""
                  class="react-calendar__navigation__arrow react-calendar__navigation__next2-button"
                  type="button"
                >
                  »
                </button>
              </div>
              <div
                class="react-calendar__viewContainer"
              >
                <div
                  class="react-calendar__month-view "
                >
                  <div
                    style="display: flex; align-items: flex-end;"
                  >
                    <div
                      style="flex-grow: 1; width: 100%;"
                    >
                      <div
                        class="react-calendar__month-view__weekdays"
                        style="display: flex; flex-wrap: no-wrap;"
                      >
                        <div
                          class="react-calendar__month-view__weekdays__weekday"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                        >
                          <abbr
                            aria-label="Monday"
                            title="Monday"
                          >
                            Mon
                          </abbr>
                        </div>
                        <div
                          class="react-calendar__month-view__weekdays__weekday"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                        >
                          <abbr
                            aria-label="Tuesday"
                            title="Tuesday"
                          >
                            Tue
                          </abbr>
                        </div>
                        <div
                          class="react-calendar__month-view__weekdays__weekday"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                        >
                          <abbr
                            aria-label="Wednesday"
                            title="Wednesday"
                          >
                            Wed
                          </abbr>
                        </div>
                        <div
                          class="react-calendar__month-view__weekdays__weekday"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                        >
                          <abbr
                            aria-label="Thursday"
                            title="Thursday"
                          >
                            Thu
                          </abbr>
                        </div>
                        <div
                          class="react-calendar__month-view__weekdays__weekday"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                        >
                          <abbr
                            aria-label="Friday"
                            title="Friday"
                          >
                            Fri
                          </abbr>
                        </div>
                        <div
                          class="react-calendar__month-view__weekdays__weekday"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                        >
                          <abbr
                            aria-label="Saturday"
                            title="Saturday"
                          >
                            Sat
                          </abbr>
                        </div>
                        <div
                          class="react-calendar__month-view__weekdays__weekday"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                        >
                          <abbr
                            aria-label="Sunday"
                            title="Sunday"
                          >
                            Sun
                          </abbr>
                        </div>
                      </div>
                      <div
                        class="react-calendar__month-view__days"
                        style="display: flex; flex-wrap: wrap;"
                      >
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 1, 2022"
                          >
                            1
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 2, 2022"
                          >
                            2
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 3, 2022"
                          >
                            3
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 4, 2022"
                          >
                            4
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 5, 2022"
                          >
                            5
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 6, 2022"
                          >
                            6
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 7, 2022"
                          >
                            7
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 8, 2022"
                          >
                            8
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 9, 2022"
                          >
                            9
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 10, 2022"
                          >
                            10
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 11, 2022"
                          >
                            11
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 12, 2022"
                          >
                            12
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 13, 2022"
                          >
                            13
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 14, 2022"
                          >
                            14
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 15, 2022"
                          >
                            15
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 16, 2022"
                          >
                            16
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__tile--now react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 17, 2022"
                          >
                            17
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 18, 2022"
                          >
                            18
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 19, 2022"
                          >
                            19
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 20, 2022"
                          >
                            20
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 21, 2022"
                          >
                            21
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 22, 2022"
                          >
                            22
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 23, 2022"
                          >
                            23
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 24, 2022"
                          >
                            24
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 25, 2022"
                          >
                            25
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 26, 2022"
                          >
                            26
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 27, 2022"
                          >
                            27
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 28, 2022"
                          >
                            28
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 29, 2022"
                          >
                            29
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 30, 2022"
                          >
                            30
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="August 31, 2022"
                          >
                            31
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="September 1, 2022"
                          >
                            1
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="September 2, 2022"
                          >
                            2
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="September 3, 2022"
                          >
                            3
                          </abbr>
                        </button>
                        <button
                          class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                          style="flex-basis: 14.285714285714286%; flex-shrink: 0; flex-grow: 0; overflow: hidden;"
                          type="button"
                        >
                          <abbr
                            aria-label="September 4, 2022"
                          >
                            4
                          </abbr>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div>
        <h4>
          Expenses:
        </h4>
        <div>
          c test details $120 on 2022/08/05T00:00
          <span>
             
          </span>
          <button>
            edit
          </button>
        </div>
        <div>
          a test details $130 on 2022/08/03T00:00
          <span>
             
          </span>
          <button>
            edit
          </button>
        </div>
        <div>
          b test description details $140 on 2022/08/01T00:00
          <span>
             
          </span>
          <button>
            edit
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
